### ISSUE TITLE
Wrong maximum length of route metadata

### ISSUE THREAD

**Issue by GooDer** [rsocket/rsocket-py#170]
I encountered problem in our project where we are using rsocket to communication with Java server.
We have for some legacy reason very long URL path to Java endpoint which is around 138 characters long.
```python
>           serialized += struct.pack('>b', len(tag))
E           struct.error: byte format requires -128 <= number <= 127
```
In this case the rsocket throw error on preparation of length byte for the tag. From the code I believe that the intention was to allow size of 0 - 255 but right now it is -128 to 127 and that raise the error. Also length check was wrong in this case since we have 256 "values" with zero included but maximum length is still 255.

## Expected Behavior
Metadata tag should allow to be up to 255 character long

## Actual Behavior
Maximum metadata length is now 127 characters

## Steps to Reproduce
See unit test in #169

## Possible Solution
Already created PR with fix #169

## Your Environment
Tried on Linux with Python 3.8 and Windows with Python 3.11

* RSocket version(s) used: 0.4.11

**Comment by jell-o-fishi** [comment]
Thank you for the PR.
If you have the time please make the changes so the tests you added would be consistent with the rest of the tests.

**Comment by jell-o-fishi** [comment]
released 0.4.12
